x/marketmap/types/aggregate_market.go

```diff
@@ -0,0 +1,93 @@
+package types
+
+import (
+	fmt "fmt"
+)
+
+// NewAggregateMarketConfig returns a new AggregateMarketConfig instance. The
+// AggregateMarketConfig represents the global set of market configurations for
+// all providers that will be utilized off-chain as well as how tickers will be
+// resolved to a final price. Each ticker can have a list of convertable markets
+// that will be used to convert the prices of a set of tickers to a common
+// ticker.
+//
+// Price aggregation broadly follows the following steps:
+//  1. Fetch prices for each ticker from the providers.
+//  2. Calculate the final price for each ticker. This is dependent on the
+//     aggregation
+//     strategy used by the oracle. The oracle may use a median price, a
+//     weighted average price, etc.
+//  3. Convert the price of each ticker to a common ticker using the aggreagted
+//     ticker
+//     configurations by default. The oracle may use a different aggregation
+//     strategy to convert the price of a ticker to a common ticker.
+//
+// For example, the oracle may be configured with the feeds:
+//   - BTC/USDT
+//   - USDT/USD
+//   - BTC/USDC
+//   - USDC/USD
+//
+// The aggregated ticker may be:
+//   - BTC/USD: (calculate a median price from the following convertable
+//     markets)
+//     1. BTC/USDT -> USDT/USD = BTC/USD
+//     2. BTC/USDC -> USDC/USD = BTC/USD
+func NewAggregateMarketConfig(markets map[string]MarketConfig, tickers map[string]PathsConfig) (AggregateMarketConfig, error) {
+	c := AggregateMarketConfig{
+		MarketConfigs: markets,
+		TickerConfigs: tickers,
+	}
+
+	if err := c.ValidateBasic(); err != nil {
+		return AggregateMarketConfig{}, err
+	}
+
+	return c, nil
+}
```

Comment:
```suggestion
	return c, c.ValidateBasic()
}
```

Reply:
imo idt we should return the struct if its not valid.

Reply:
fp